src/PropertySuggester/Suggesters/SimpleSuggester.php

```diff
@@ -172,14 +172,15 @@ public function suggestByItem( Item $item, $limit, $minProbability, $context ) {
 
 				$entityId = $dataValue->getEntityId();
 
-				if ( !( $entityId instanceof Int32EntityId ) ) {
+				if ( !( $entityId instanceof ItemId ) ) {
```

Comment:
This code can not consume IDs of different types, because the type information is entirely dropped. What is stored in the database are numbers. When read, the code consuming this assumes the numbers refer to Items. It's really important to make sure no other type gets accidentally be saved in the same table.